widget/entry.go

```diff
@@ -999,6 +999,18 @@ func (e *Entry) updateText(text string) {
 		}
 	})
 
+	if validate := e.Validator; validate != nil {
+		if err := validate(text); err != e.validationError {
+			e.validationError = err
+
+			if f := e.onValidationChanged; f != nil {
+				f(err)
+			}
+		}
+
+		e.validationStatus.Refresh()
```

Comment:
Would this be simpler?
```suggestion
			e.setValidationError(err)
		} else {
			e.validationStatus.Refresh()
```

Reply:
Thanks for the suggestion. I slimmed things down even more :)